Sensors pcb

Bobby-Watson opened the pull request.

efondere committed:
sensors: create project from template

Created the KiCad projet from the sensors PCB from the latest version of
the template (parent to this commit). Author, contributor and reviewer
names were removed, title and date were changed from the drawing title
block.

malapravong committed:
Création du symbole pour le IMU

Bobby-Watson committed:
Ajusté symbole IMU et commencé schéma

Bobby-Watson committed:
sensors-pcb: adjust IMU schematic and symbol

hide no connect pins
add connect I2C I/O to STM32 with pull-up resistors
add external 42kHz clock from Exocet template
pins 15 and 16 connected to GND through solder jumper
because of conflicting datasheet info (2021 vs 2016). to be tested IRL

Bobby-Watson committed:
sensors-pcb: add second solder junction

in case connecting pins together is illegal

malapravong committed:
Symbole GPS

Bobby-Watson committed:
sensors-pcb: finish GPS schematic

malapravong committed:
sensors-pcb: add footprints

Bobby-Watson committed:
sensors-pcb: clean up and fix schematic

Bobby-Watson committed:
sensors-pcb: route IMU

Bobby-Watson committed:
sensors-pcb: readded ground plane

oups :P

Bobby-Watson committed:
sensors-pcb: add coax connector, link JLCPCB numbers

efondere commented on sensors-pcb/sensors-pcb.kicad_sch:
On devrait ajouter un condensateur de découplage de 100n sur la pin VDDIO (copier ceux du STM32 au besoin).

Bobby-Watson replied:
Tu veux bien dire le VDDIO du IMU? Si oui, c'est pas correct d'utiliser le même que pour le VDD? Dans tous les cas, est-ce qu'il faudrait en mettre un pour le GPS même si la spec le demande pas?

efondere replied:
Oups... désolé j'ai écrit ce commentaire y'a un bout c'est vrai qu'il est pas très clair. Je parlais effectivement du IMU, mais, en regardant le footprint, les deux pins sont assez proches alors effectivement, on peut sûrement le partager (voir mon autre commentaire sur le condensateur C1 justement).

Pour le GPS, on ne perd rien à en ajouter, et, selon l'emplacement des pins d'alimentation sur le footprint, on peut décider de le partager comme pour le IMU ou pas. Tu peux prendre le même condensateur aussi.

efondere commented on sensors-pcb/sensors-pcb.kicad_sch:
Par symmétrie, vous pouvez ajouter des résistances pull-up sur les SCL et SDA du GPS. Vous pouvez utiliser les mêmes que pour le IMU.

Bobby-Watson replied:
Ils sont built-in selon la datasheet, j'imagine qu'il en faut donc pas?

efondere replied:
Si c'est built-in dans le GPS, alors effectivement il faut pas en mettre en double!

efondere commented on sensors-pcb/sensors-pcb.kicad_sch:
Il faudrait ajouter des test points sur les nets suivants:
- SCL et SDA pour GPS et IMU
- nRESET sur le IMU
- (facultatif) RESET et SAFEBOOT pour le GPS

Copier ceux de la page du STM32 pour les numéros JLCPCB. Garder en do not populate.

Ajouter le nom des nets à côté des TP sur le Silkscreen du PCB.

efondere commented on sensors-pcb/sensors-pcb.kicad_pro:
FACULATIF: Mettre à jour les metadata du PCB.

En double-cliquant sur la cartouche en bas à droite, vous pourrez modifier:
- La date
- Le titre de la page (laissez tel-quel)
- La version (on n'a pas encore décidé d'une bonne façon de versionner les PCB, donc je vous laisse choisir)

Il y aura peut-être à faire ça pour le schéma et pour le PCB si il ne se met pas à jour automatiquement.

En cliquant sur le 2e bouton à partir de la gauche (à côté de la disquette) du menu d'outil horizontal en haut de KiCad, ça ouvre la fenêtre "schematic setup". Allez dans "Project/Text Variables" puis vous pourrez changer:
- L'auteur (une ou plusieurs personnes)
- Les contributeurs
- Les réviseurs (mettez mon nom)

Laissez le code du projet et le nom comme ils sont.

Finalement, vous pouvez modifier le texte avec le nom du PCB pour matcher ce qu'il va y avoir dans la prochaine version du template:

<img width="644" height="445" alt="Image" src="https://github.com/user-attachments/assets/b8f7c24d-a9e8-4b67-b6f4-1e478dad74f5" />

Tout ça est purement esthétique, donc pas nécessaire pour commander le PCB donc ne perdez pas votre temps avec ça!

efondere commented on sensors-pcb/sensors-pcb.kicad_pcb:
C1 semble être un condensateur de découplage. Il faudrait donc le mettre près des pins +3.3V du BNO05, comme ça:

<img width="972" height="733" alt="Image" src="https://github.com/user-attachments/assets/0f68a32f-3aab-4ee0-a213-990d450ec48d" />

(code de couleur: bleu choses à déplacer, vert: ce qu'on devrait voir à la fin, rose: ce qui ne devrait plus être là à la fin)

On peut au passage bouger le via pour que le courant passe dans cet ordre: via -> C1 -> pins du BNO05.

efondere commented on sensors-pcb/sensors-pcb.kicad_pcb:
Les footprints utilisés pour les solder bridges sont actuellement _bridged_ (connectés par défaut). Il faut donc les changer pour les footprints "Open" (ex. `SolderJumper-2_P1.3mm_Open_RoundedPad1.0x1.5mm`).

efondere commented on sensors-pcb/sensors-pcb.kicad_sch:
Les traces pour SCL et SDA du BNO05 ne sont pas de la même longueur. Il faut utiliser l'outil "Tune length of single track" pour alonger la plus courte des deux.

<img width="137" height="173" alt="Image" src="https://github.com/user-attachments/assets/8f49cebb-73a3-44cd-b71b-73ed39f616a1" />

efondere commented on sensors-pcb/sensors-pcb.kicad_pcb:
Les traces pour SCL et SDA du BNO05 ne sont pas de la même longueur. Il faut utiliser l'outil "Tune length of single track" pour alonger la plus courte des deux.

<img width="137" height="173" alt="Image" src="https://github.com/user-attachments/assets/8f49cebb-73a3-44cd-b71b-73ed39f616a1" />

efondere commented on sensors-pcb/sensors-pcb.kicad_pcb:
Il reste quelques warnings en roulant le DRC au niveau du Silkscreen qui peuvent être réglées en déplaçant légèrement les références de certaines pièces.

efondere commented on sensors-pcb/sensors-pcb.kicad_sch:
Les solder jumper devraient être en "Do not populate" (clic droit -> Attributes -> cocher "Do not Populate") pour s'assurer que JLCPCB ne les soude pas à la fabrication.

efondere commented on sensors-pcb/sensors-pcb.kicad_sch:
La résistance R5 de 40k n'est pas basic ou "extended avec un pouce". Il faudrait prendre une résistance d'une valeur proche, mais standard (E24, voir le calculateur de KiCad) qui sera basic sur JLCPCB, par exemple 39k (qui est extended avec un pouce, donc équivalent à basic pour nous): https://jlcpcb.com/partdetail/26526-0402WGF3902TCE/C25783

efondere commented on sensors-pcb/sensors-pcb.kicad_sch:
À valider si ça affecte beaucoup le prix lors de l'impression (donc ignorer ceci pour l'instant), mais il faudrait possiblemenent remplacer l'inductance L1 de 27nH pour utiliser des basic/extended avec pouce. Voici des pièces de remplacement (ll faudrait les mettre en série):
- 15 nH: https://jlcpcb.com/partdetail/Sunlord-SDCL1005C15NJTDF/C27143
- 10 nH: https://jlcpcb.com/partdetail/Sunlord-SDCL1005C10NJTDF/C27147

efondere reviewed (changes requested):
Il reste quelques petits changements, après on sera prêt à commander!

Bobby-Watson committed:
sensors-pcb: fix the stuff

Ajouté des test points
Ajusté la longueur de la trace SCL_IMU
Ajouté condensateur de découplage pour le GPS
Mis a jour la metadata
Changé les solder jumpers
Reglé problèmes DRC. Il en reste 3 mais c'est lié au template et je sais pas trop comment les regler

Bobby-Watson committed:
sensors-pcb: update resistor

J'avais déjà associé la résistance 40k avec un code JLCPCB pour une 40.2k et elle semble être accessible donc voilà

efondere committed:
sensors-pcb: change R5 for basic JLCPCB part

efondere committed:
sensors-pcb: set TPs solder junctions as DNP

efondere committed:
sensors-pcb: update PCB metadata and info block

efondere committed:
sensors-pcb: connector SDA_GPS test point

Bobby-Watson committed:
sensors-pcb: replace IMU with MPU-6050

Bobby-Watson committed:
sensors-pcb: change 27nH to 10nH+15nH

Bobby-Watson committed:
sensors-pcb: change antenna coax connector

Bobby-Watson committed:
sensors-pcb: add test point for fuse

Bobby-Watson committed:
sensors-pcb: add IMU address text

Bobby-Watson committed:
sensors-pcb: add pull-up resistors

Bobby-Watson committed:
sensors-pcb: fix coax connector footprint

Bobby-Watson committed:
sensors-pcb: change fuse to DNP

Bobby-Watson committed:
sensors-pcb: ghost busted

fixed fuse test point label

gumile committed:
sensors-pcb: add fuse holder 01110501Z

efondere committed:
sensors-pcb: finalize v1.1.0 metadata

Added Guillaume as reviewer and updated the date to the last
modification.

Also added missing JLCPCB part number to the MPU-6050 symbol.

efondere committed:
sensors-pcb: update LED series resistor value

efondere committed:
sensors-pcb: add some 3d-models
